poller/poller.go

```diff
@@ -136,16 +144,56 @@ func (poller *Poller) jobs() {
 	}
 }
 
+// getSerializedDataSubset returns serialized data for the splits provided
+func (poller *Poller) getSerializedDataSubset(splits []string) string {
+	currentSplitData := poller.getSplitData()
+	updatedSubsets := poller.getCachedSerializedDataSubsets()
+	sort.Strings(splits)
+	key := strings.Join(splits, ".")
+
+	subset, inMap := updatedSubsets[key]
+	if inMap {
+		return subset
+	}
+	subset = generateSerializedData(currentSplitData, splits)
+	updatedSubsets[key] = subset
+
+	// update cache
+	updatedCache := Cache{
+		splitData:             currentSplitData,
+		serializedData:        poller.getSerializedData(),
+		serializedDataSubsets: updatedSubsets,
+	}
+	atomic.StorePointer(&poller.cache, unsafe.Pointer(&updatedCache))
+
+	return subset
+}
+
+// getUpdatedSerializedDataSubsets updates cached serializedDataSubsets based on new split data
+func (poller *Poller) getUpdatedSerializedDataSubsets(newSplitData SplitData) map[string]string {
+	updatedSubsets := poller.getCachedSerializedDataSubsets()
+	for key := range updatedSubsets {
+		updatedSubsets[key] = generateSerializedData(newSplitData, strings.Split(key, "."))
+	}
+	return updatedSubsets
+}
+
 // generateSerializedData takes SplitData and generates a script tag
 // that saves the SplitData info to the window object of the browser
-func generateSerializedData(splitData SplitData) string {
+func generateSerializedData(splitData SplitData, splits []string) string {
 	if reflect.DeepEqual(splitData, SplitData{}) {
 		return emptyCacheLoggingScript
 	}
 	splitsData := map[string]string{}
 
 	// Serialize values for splits
 	for _, split := range splitData.Splits {
+		splitIndex := sort.SearchStrings(splits, split.Name)
+		splitInSplits := splitIndex < len(splits) && splits[splitIndex] == split.Name
```

Comment:
Can you rethink the variable names in this function?
We have splitData, splitsData, splits, splitInSplits. Boggling my mind a bit trying to read through it